RNTester/js/components/RNTesterExampleList.js

```diff
@@ -207,36 +211,126 @@ const renderSectionHeader = ({section}) => (
+    super(props);
+    this.state = {
+      components: props.list.ComponentExamples,
+      api: props.list.APIExamples,
+      recentComponents: [],
+      recentApis: [],
+    };
+  }
+
+  componentDidMount() {
+    AsyncStorage.getItem('RecentComponents', (err, storedString) => {
+      if (err || !storedString) {
+        return;
+      }
+      const recentComponents = JSON.parse(storedString);
+      this.setState({
+        recentComponents: recentComponents,
+      });
+    });
+    AsyncStorage.getItem('RecentApi', (err, storedString) => {
+      if (err || !storedString) {
+        return;
+      }
+      const recentApis = JSON.parse(storedString);
+      this.setState({
+        recentApis: recentApis,
+      });
+    });
+  }
+
+  updateSectionsList = (index, key) => {
+    if (key === 'Components') {
+      let openedItem = this.state.components[index];
+      let componentsCopy = [...this.state.recentComponents];
+      const ind = componentsCopy.findIndex(component => component.key === openedItem.key);
+      if(ind != -1) {
+        componentsCopy.splice(ind, 1);
+      }
+      if (this.state.recentComponents.length >= 5) {
+        componentsCopy.pop();
+      }
+      componentsCopy.unshift(openedItem);
+      AsyncStorage.setItem('RecentComponents', JSON.stringify(componentsCopy));
+    } else {
+      let openedItem = this.state.api[index];
+      let apisCopy = [...this.state.recentApis];
+      const ind = apisCopy.findIndex(api => api.key === openedItem.key);
+      if(ind != -1) {
+        apisCopy.splice(ind, 1);
+      }
+      if (this.state.recentApis.length >= 5) {
+        apisCopy.pop();
+      }
+      apisCopy.unshift(openedItem);
+      AsyncStorage.setItem('RecentApi', JSON.stringify(apisCopy));
+    }
+  };
+
   render(): React.Node {
     const filter = ({example, filterRegex, category}) =>
       filterRegex.test(example.module.title) &&
       (!category || example.category === category) &&
       (!Platform.isTV || example.supportsTVOS);
 
-      const {screen} = this.props; 
-      let sections = []; 
-      if (screen === "component"){ 
+    const {screen} = this.props;
+    let sections = [];
+    if (screen === 'component') {
+      if (this.state.recentComponents.length > 0) {
         sections = [
           {
-            data: this.props.list.ComponentExamples,
+            data: this.state.recentComponents,
+            key: 'Component',
+            title: 'Recently viewed'
+          },
+          {
+            data: this.state.components,
             key: 'Components',
-          }
+            title: 'Components'
+          },
+        ];
+      } else {
+        sections = [
+          {
+            data: this.state.components,
+            key: 'Components',
+          },
+        ];
+      }
+    } else if (screen === 'api') {
+      if (this.state.recentApis.length > 0) {
+        sections = [
+          {
+            data: this.state.recentApis,
+            key: 'API',
```

Comment:
Everything looks good. Maybe, you can try naming the key to `RECENT_APIS` so that makes a little more sense when you are doing

```js
if (key == 'APIS' && key === 'RECENT_APIS') {

}
```

Reply:
Done!